tap-agent/src/agent/sender_account.rs

```diff
@@ -484,12 +493,23 @@ impl Actor for SenderAccount {
                     }
                 }
 
-                // Maybe allow the sender right after the potential RAV request. This way, the
-                // sender can be allowed again as soon as possible if the RAV was successful.
-
-                let should_allow = state.denied && !state.deny_condition_reached();
-                if should_allow {
-                    state.remove_from_denylist().await;
+                match (state.denied, state.deny_condition_reached()) {
+                    // Maybe allow the sender right after the potential RAV request. This way, the
+                    // sender can be allowed again as soon as possible if the RAV was successful.
```

Comment:
If this comment is about the `(true, false)` case, then it *will* allow the sender.
```suggestion
                    // Allow the sender right after the potential RAV request. This way, the
                    // sender can be allowed again as soon as possible if the RAV was successful.
```